azext_iot/common/shared.py

```diff
@@ -304,5 +309,33 @@ class SHAHashVersions(Enum):
     """
     Supported SHA types for generating the certificate thumbprint.
     """
+
     SHA1 = 1
     SHA256 = 256
+
+
+# Utility classes for nested edge config file values and device arguments
+class EdgeContainerAuth(NamedTuple):
+    serveraddress: Optional[str] = None
+    username: Optional[str] = None
+    password: Optional[str] = None
+
+
+class NestedEdgeDeviceConfig(NamedTuple):
+    device_id: str
```

Comment:
The minimum operation with just a device Id is to create an edge enabled device, is that right?

Reply:
Yes, correct, technically that is the only property you need in the config to create devices.
For this scenario, they're all assumed edge devices, but this could easily be `DeviceConfig` with a property of `edge: bool`